auslib/test/admin/views/test_releases.py

```diff
@@ -783,3 +800,68 @@ def testSettings(self):
         ret = self._get('/releases/settings')
         self.assertStatusCode(ret, 200)
         self.assertEqual(json.loads(ret.data), json.loads(data['blob']))
+
+
+class TestReadOnlyView(ViewTest, JSONTestMixin):
+
+    def testReadOnlyGet(self):
+        ret = self._get('/releases/b/read_only')
+        is_read_only = dbo.releases.t.select(dbo.releases.name == 'b').execute().first()['read_only']
+        self.assertEqual(json.loads(ret.data)['read_only'], is_read_only)
+
+    def testReadOnlySetTrueAdmin(self):
+        data = dict(name='b', read_only=True, product='Firefox', data_version=1)
+        self._put('/releases/b/read_only', username='bill', data=data)
+        read_only = dbo.releases.isReadOnly(name='b')
+        self.assertEqual(read_only, True)
```

Comment:
While testing with the UI and some manual requests to the API I realized that these tests are not quite valid....you're passing a Python dict rather than JSON data. Although the tests pass, they don't simulate real requests. You should dump the data to json before passing it, like we do in other tests, eg: https://github.com/mozilla/balrog/blob/master/auslib/test/admin/views/test_releases.py#L27

You should end up with some failures like this after doing so:
__________________________________________________________________________________________________________________ TestReadOnlyView.testReadOnlySetTrueAdmin __________________________________________________________________________________________________________________

self = <auslib.test.admin.views.test_releases.TestReadOnlyView testMethod=testReadOnlySetTrueAdmin>

    def testReadOnlySetTrueAdmin(self):
        data = json.dumps(dict(name='b', read_only=True, product='Firefox', data_version=1))
        self._put('/releases/b/read_only', username='bill', data=data)
        read_only = dbo.releases.isReadOnly(name='b')
>       self.assertEqual(read_only, True)
E       AssertionError: False != True

auslib/test/admin/views/test_releases.py:816: AssertionError


Sorry for not catching this earlier...shouldn't be too difficult to fix though.